pokete_classes/npcs.py

```diff
@@ -0,0 +1,168 @@
+
+    def check_walk(self, x, y):
+        """Checks whether or not the NPC can walk to a point"""
+        vec = se.Line(" ", x - self.x, y - self.y)
+        return not any([any(j.state == "solid"
+                    for j in self.map.obmap[i.ry + self.y][i.rx + self.x])
+                        for i in vec.obs][1:])
+
+    def walk_point(self, x, y):
+        """Walks the NPC tp a certain point"""
+        o_x = self.x
+        o_y = self.y
+        vec = se.Line(" ", x - o_x, y - o_y)
+        if not self.check_walk(x, y):
+            return False
+        for i in vec.obs:
+            self.set(i.rx + o_x, i.ry + o_y)
+            self.mvmp.full_show()
+            time.sleep(0.2)
+        return True
+
+    def give(self, name, item):
+        """Method thats gifts an item to the player"""
+        item = getattr(self.invitems, item)
+        self.will = False
+        self.used_npcs.append(self.name)
+        if ask_bool(self._ev, self.mvmp,
+                    f"{name} gifted you a '{item.pretty_name}'. \
+Do you want to accept it?"):
+            self.fig.give_item(item.name)
+
+
+class Trainer(NPC):
+    """Trauner class to fight against"""
+
+    def __init__(self, poke, name, gender, texts, lose_texts, no_poke_texts,
+                 win_texts, fight):
+        super().__init__(name, texts)
+        # attributes
+        self.gender = gender
+        self.poke = poke
+        self.lose_texts = lose_texts
+        self.no_poke_texts = no_poke_texts
+        self.win_texts = win_texts
+        self.fight = fight
+
+    def add(self, _map, x, y):
+        """Add wrapper"""
+        line = se.Line(" ", 0, _map.height, state="float")
+        for i, obj in enumerate(line.obs):
+            line.obs[i] = NPCTrigger(self)
+            line.obs[i].rx = obj.rx
+            line.obs[i].ry = obj.ry
+        line.add(_map, x, 0)
+        super().add(_map, x, y)
+
+    def action(self):
+        """Interaction with the trainer"""
+        o_x = self.x
+        o_y = self.y
+        if self.fig.has_item("shut_the_fuck_up_stone"):
+            return
+        if self.fig.x == self.x and self.poke.hp > 0 \
+                and (self.name not in self.used_npcs
+                    or not self.settings.save_trainers) \
+                and self.check_walk(self.fig.x, self.fig.y):
+            self.mvmp.full_show()
+            time.sleep(0.7)
+            self.exclamate()
+            self.walk_point(self.fig.x, self.fig.y)
+            if any(poke.hp > 0 for poke in self.fig.pokes[:6]):
+                self.text(self.texts)
+                winner = self.fight([poke for poke in self.fig.pokes[:6]
+                                    if poke.hp > 0][0],
+                               self.poke, info={"type": "duel", "player": self})
+                self.text({True: self.lose_texts,
+                           False: self.win_texts + [" < Here u go 20$"]}
+                                [winner == self.poke])
```

Comment:
```suggestion
                          [winner == self.poke])
```

Reply:
This is indented, to show that it's not another argument for `self.text`, but rather the indexation of the dict before.